cmd2.py

```diff
@@ -1266,17 +1292,14 @@ def select(self, opts, prompt='Your choice? '):
                                                                                                    len(fulloptions)))
         return result
 
-    @options([make_option('-l', '--long', action="store_true", help="describe function of parameter")])
-    def do_show(self, arg, opts):
-        """Shows value of a parameter."""
-        # If arguments are being passed as a list instead of as a string
-        if USE_ARG_LIST:
-            if arg:
-                arg = arg[0]
-            else:
-                arg = ''
-
-        param = arg.strip().lower()
+    argparser = argparse.ArgumentParser(description='show value of a parameter')
+    argparser.add_argument('-l', '--long', action='store_true', help='describe function of parameter')
+    argparser.add_argument('param', nargs='?', help='name of parameter, if not supplied show all parameters')
+    @with_argument_parser(argparser)
+    def do_show(self, arglist, args):
```

Comment:
If all of the arguments are being parsed by ``argparse`` when the **with_argument_parser** decorator is used, then why are we also passing the full arglist argument?  This seems redundant and unnecessary.

Reply:
I didn't really think about it, I just did it the same way that the **@options** one worked. But as I think about it, there is no reason to pass the argils too. You can't go backwards to the original input from the arglist, so it really is redundant and not very useful. After I finish migrating **do_save()** to use **@with_argument_list** I'll make this change.